src/cfnlint/decode/node.py

```diff
@@ -0,0 +1,179 @@
+"""
+  Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  software and associated documentation files (the "Software"), to deal in the Software
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import sys
+from copy import deepcopy
+import six
+
+
+def create_str_node_class(cls):
+    """
+    Create string node class
+    """
+    class node_class(cls):
+        """Node class created based on the input class"""
+        def __init__(self, x, start_mark, end_mark):
+            try:
+                cls.__init__(self, x)
+            except TypeError:
+                cls.__init__(self)
+            self.start_mark = start_mark
+            self.end_mark = end_mark
+
+        # pylint: disable=bad-classmethod-argument, unused-argument
+        def __new__(self, x, start_mark, end_mark):
+            if sys.version_info >= (3, 0):
+                return cls.__new__(self, x)
+
+            if isinstance(x, six.string_types):
+                return cls.__new__(self, x.encode('ascii', 'ignore'))
```

Comment:
ASCII/UTF-8

Reply:
#124 should provide some insights.

Also this part is a direct copy from what is there right now and the deleted file str_node.py